### PR TITLE
Restore option for cluster type

### PR DESCRIPTION
**Pull request by vargup**
For restore, added a new option to restore on existing cluster with same token range or a completely new cluster bootstrapped from scratch. For same cluster, use nodetool refresh to restore data and for new cluster use sstableloader. Difference is due to the significant performance disparity.

### CODE REVIEW

**Comment by triclambert** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

**Comment by mohitsoni** [comment]
@mesosphere-ci retest this please

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java [hunk 1]
```diff
@@ -237,20 +238,31 @@ public void download(BackupRestoreContext ctx) throws Exception {
...
-                for (File cfDir : getColumnFamilyDir(keyspace)) {
-                    final String columnFamily = cfDir.getName().substring(0, cfDir.getName().indexOf("-"));
-                    final Map<String, Long> snapshotFileKeys = listSnapshotFiles(amazonS3Client,
-                            bucketName,
-                            backupName + "/" + nodeId + "/" + keyspace.getName() + "/" + columnFamily);
-                    for (String fileKey : snapshotFileKeys.keySet()) {
-                        final String destinationFile = cfDir.getAbsolutePath() + fileKey.substring(fileKey.lastIndexOf("/"));
-                        downloadFile(tx, bucketName, fileKey, destinationFile);
-                        LOGGER.info("Keyspace {}, Column Family {}, FileKey {}, destination {}", keyspace, columnFamily, fileKey, destinationFile);
+            if (ctx.getRestoreType().equals("new")) {
```
Comment (mohitsoni) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java]: Nit: Use `Objects.equals()`.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java [hunk 2]
```diff
@@ -79,6 +82,16 @@ public void setAzureKey(String azureKey) {
     this.azureKey = azureKey;
   }
 
+  public String getRestoreType() {
+    if (restoreType != null && !restoreType.isEmpty()) {
```
Comment (mohitsoni) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java]: Nit: Maybe use `StringUtils.isNotBlank(restoreType)`

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java [hunk 3]
```diff
@@ -108,6 +122,10 @@ private boolean isValidAzureRequest() {
     return azureAccount != null && azureKey != null && externalLocation.startsWith("azure:");
   }
 
+  private boolean isValidRestoreType() {
+    return restoreType == null || restoreType.isEmpty() ? true: restoreType.matches("existing|new");
```
Comment (mohitsoni) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java]: Why is a null OR empty restoreType valid ?

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java (hunk 1 above)
Comment (vargup) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java]: will fix in next PR of restore schema.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java (hunk 3 above)
Comment (mohitsoni) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/BackupRestoreRequest.java]: Okay. Got it.

**Pull request merged**